connectors/source.py

```diff
@@ -501,6 +508,34 @@ async def get_file(doit=True, timestamp=None):
         """
         raise NotImplementedError
 
+    async def get_docs_incrementally(self, sync_cursor, filtering=None):
```

Comment:
Not sure about the method name - maybe just `get_docs_incremental`? `get_docs_from_cursor`?

Also I can see that Full Sync can also benefit from cursor especially when sync was suspended. Do you wanna support it now or later?

Reply:
> Not sure about the method name - maybe just `get_docs_incremental`? `get_docs_from_cursor`?

I'm open to renaming it. WDYT `get_delta_docs`?

> Also I can see that Full Sync can also benefit from cursor especially when sync was suspended. Do you wanna support it now or later?

Not for now, see this comment: https://github.com/elastic/connectors-python/pull/986#discussion_r1211961681.

Reply:
I think `get_docs_incrementally` is a bit better, I guess let's keep it like that